src/X12.Parsing/X12StreamReader.cs

```diff
@@ -1,229 +1,231 @@
-                {
-                    sb.Append(one);
-                }
-
-                if (isBinary && one[0] == this.Delimiters.ElementSeparator)
-                {
-                    int binarySize = 0;
-                    string[] elements = sb.ToString().Split(this.Delimiters.ElementSeparator);
-                    if (elements[0] == "BIN" && elements.Length >= 2)
-                    {
-                        int.TryParse(sb.ToString().Split(this.Delimiters.ElementSeparator)[1], out binarySize);
-                    }
-
-                    if (elements[0] == "BDS" && elements.Length >= 3)
-                    {
-                        int.TryParse(sb.ToString().Split(this.Delimiters.ElementSeparator)[2], out binarySize);
-                    }
-
-                    if (binarySize > 0)
-                    {
-                        var buffer = new char[binarySize];
-                        this.reader.Read(buffer, 0, binarySize);
-                        sb.Append(buffer);
-                        break;
-                    }
-                }
-
-                if (!isBinary && (sb.ToString() == "BIN" + this.Delimiters.ElementSeparator 
-                                  || sb.ToString() == "BDS" + this.Delimiters.ElementSeparator))
-                {
-                    isBinary = true;
-                }
-            }
-
-            return sb.ToString().TrimStart();
+          isBinary = true;
         }
+      }
 
-        /// <summary>
-        /// Returns the next transaction read from the stream. If no transaction is read, then null is returned
-        /// </summary>
-        /// <returns>Transaction read from the stream, if found; otherwise, null</returns>
-        public X12FlatTransaction ReadNextTransaction()
-        {
-            var segments = new StringBuilder();
+      return sb.ToString().TrimStart();
+    }
 
-            string segmentString;
-            string segmentId;
-            do
+    /// <summary>
+    /// Returns the next transaction read from the stream. If no transaction is read, then null is returned
+    /// </summary>
+    /// <returns>Transaction read from the stream, if found; otherwise, null</returns>
+    public X12FlatTransaction ReadNextTransaction()
+    {
+      var segments = new StringBuilder();
+
+      string segmentString;
+      string segmentId;
+      do
+      {
+        segmentString = this.ReadNextSegment();
+        segmentId = this.ReadSegmentId(segmentString);
+        switch (segmentId)
+        {
+          case "ISA":
+            this.CurrentIsaSegment = segmentString + this.Delimiters.SegmentTerminator;
+            break;
+          case "GS":
+            this.CurrentGsSegment = segmentString + this.Delimiters.SegmentTerminator;
+            break;
+          case "IEA":
+          case "GE":
+            break;
+          case null:
+            // This case is added to handle where EOF is reached with no segments read.
+            break;
```

Comment:
Unfortunately, this was hit with a lot of minor formatting changes that don't actually change the file. This bit is the core of this PR, where we catch the case where EOF is hit before we read any segments.